File: решения/java/src/Lab14/task3956example/Shape.java
package Lab14.task3956example;

/**
 * @author akryukov
 *         17.07.2017
 */
public abstract class Shape {
    private String stroke;
    private int strokeWidth;
    private String fill;

    public Shape(String stroke, int strokeWidth, String fill) {
        this.stroke = stroke;
        this.strokeWidth = strokeWidth;
        this.fill = fill;
    }

    public String getStroke() {
        return stroke;
    }

    public int getStrokeWidth() {
        return strokeWidth;
    }

    public String getFill() {
        return fill;
    }

    public abstract void slide(int dx, int dy);

    public abstract Point getCenter();

    public abstract double area();
}

File: решения/cs/Classes/task3956example/Shape.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Classes.Task3956example
{
    public abstract class Shape
    {
        public String Stroke { private set; get; }
        public int StrokeWidth { private set; get; }
        public String Fill { private set; get; }
        public abstract double Area { get; }

        public Shape(String stroke, int strokeWidth, String fill)
        {
            Stroke = stroke;
            StrokeWidth = strokeWidth;
            Fill = fill;
        }

        public abstract void Slide(int dx, int dy);

        public abstract Point MakeCenter();
    }
}
